lib/internal/http2/compat.js

```diff
@@ -181,6 +181,14 @@ class Http2ServerRequest extends Readable {
     return '2.0';
   }
 
+  get socket() {
+    return this.stream.session.socket;
+  }
+
+  get connection() {
+    return this.socket;
```

Comment:
One point on this: we *really* need to discourage direct access to the socket. With an http/2 connection, bad things can happen if the user does not know what they are doing. The underlying native Http2Session class takes complete ownership of reading and writing to the socket and any direct interactions with the socket by user code could throw the entire connection out of sync. I'd prefer not to expose these at all but if we need to, let's add a strong warning in the docs.